From 34cf7af0e250775a9446ab9fba1ead290cade18f Mon Sep 17 00:00:00 2001
From: Gurpreet Ghai <gghai@codeaurora.org>
Date: Fri, 15 Jun 2018 15:36:05 +0530
Subject: [PATCH 03/10] BT: Adding APTX-Adaptive codec to BT HAL

Adding APTX-Adaptive codec entry to codec index.

CRs-Fixed: 2280216

Change-Id: I73ab0792a353b546338b2134d946ec500f7308f2
---
 include/hardware/bt_av.h        | 4 ++++
 stack/a2dp/a2dp_codec_config.cc | 2 ++
 stack/a2dp/a2dp_vendor.cc       | 4 ++++
 stack/test/stack_a2dp_test.cc   | 6 ++++++
 4 files changed, 16 insertions(+)

diff --git a/include/hardware/bt_av.h b/include/hardware/bt_av.h
index c4c0ecc82..12e890327 100644
--- a/include/hardware/bt_av.h
+++ b/include/hardware/bt_av.h
@@ -53,6 +53,7 @@ typedef enum {
   BTAV_A2DP_CODEC_INDEX_SOURCE_AAC,
   BTAV_A2DP_CODEC_INDEX_SOURCE_APTX,
   BTAV_A2DP_CODEC_INDEX_SOURCE_APTX_HD,
+  BTAV_A2DP_CODEC_INDEX_SOURCE_APTX_ADAPTIVE,
   BTAV_A2DP_CODEC_INDEX_SOURCE_LDAC,
 
   BTAV_A2DP_CODEC_INDEX_SOURCE_MAX,
@@ -146,6 +147,9 @@ typedef struct {
       case BTAV_A2DP_CODEC_INDEX_SOURCE_APTX_HD:
         codec_name_str = "aptX HD";
         break;
+      case BTAV_A2DP_CODEC_INDEX_SOURCE_APTX_ADAPTIVE:
+        codec_name_str = "aptX Adaptive";
+        break;
       case BTAV_A2DP_CODEC_INDEX_SOURCE_LDAC:
         codec_name_str = "LDAC";
         break;
diff --git a/stack/a2dp/a2dp_codec_config.cc b/stack/a2dp/a2dp_codec_config.cc
index edf7e0c46..891583168 100644
--- a/stack/a2dp/a2dp_codec_config.cc
+++ b/stack/a2dp/a2dp_codec_config.cc
@@ -128,6 +128,8 @@ A2dpCodecConfig* A2dpCodecConfig::createCodec(
     case BTAV_A2DP_CODEC_INDEX_SOURCE_APTX_HD:
       codec_config = new A2dpCodecConfigAptxHd(codec_priority);
       break;
+    case BTAV_A2DP_CODEC_INDEX_SOURCE_APTX_ADAPTIVE:
+      break;
     case BTAV_A2DP_CODEC_INDEX_SOURCE_LDAC:
       codec_config = new A2dpCodecConfigLdacSource(codec_priority);
       break;
diff --git a/stack/a2dp/a2dp_vendor.cc b/stack/a2dp/a2dp_vendor.cc
index bcea13d1c..991d822c8 100644
--- a/stack/a2dp/a2dp_vendor.cc
+++ b/stack/a2dp/a2dp_vendor.cc
@@ -595,6 +595,8 @@ const char* A2DP_VendorCodecIndexStr(btav_a2dp_codec_index_t codec_index) {
       return A2DP_VendorCodecIndexStrAptx();
     case BTAV_A2DP_CODEC_INDEX_SOURCE_APTX_HD:
       return A2DP_VendorCodecIndexStrAptxHd();
+    case BTAV_A2DP_CODEC_INDEX_SOURCE_APTX_ADAPTIVE:
+      break;
     case BTAV_A2DP_CODEC_INDEX_SOURCE_LDAC:
       return A2DP_VendorCodecIndexStrLdac();
     case BTAV_A2DP_CODEC_INDEX_SINK_LDAC:
@@ -620,6 +622,8 @@ bool A2DP_VendorInitCodecConfig(btav_a2dp_codec_index_t codec_index,
       return A2DP_VendorInitCodecConfigAptx(p_cfg);
     case BTAV_A2DP_CODEC_INDEX_SOURCE_APTX_HD:
       return A2DP_VendorInitCodecConfigAptxHd(p_cfg);
+    case BTAV_A2DP_CODEC_INDEX_SOURCE_APTX_ADAPTIVE:
+      break;
     case BTAV_A2DP_CODEC_INDEX_SOURCE_LDAC:
       return A2DP_VendorInitCodecConfigLdac(p_cfg);
     case BTAV_A2DP_CODEC_INDEX_SINK_LDAC:
diff --git a/stack/test/stack_a2dp_test.cc b/stack/test/stack_a2dp_test.cc
index fecb3cada..ed8e457d4 100644
--- a/stack/test/stack_a2dp_test.cc
+++ b/stack/test/stack_a2dp_test.cc
@@ -172,6 +172,7 @@ const uint8_t codec_info_non_a2dp_dummy[AVDT_CODEC_SIZE] = {
 
 static const char* APTX_ENCODER_LIB_NAME = "libaptX_encoder.so";
 static const char* APTX_HD_ENCODER_LIB_NAME = "libaptXHD_encoder.so";
+static const char* APTX_ADAPTIVE_ENCODER_LIB_NAME = "libaptXAdaptive_encoder.so";
 static const char* LDAC_ENCODER_LIB_NAME = "libldacBT_enc.so";
 static const char* LDAC_DECODER_LIB_NAME = "libldacBT_dec.so";
 
@@ -213,6 +214,11 @@ class StackA2dpTest : public ::testing::Test {
           // shared library installed.
           supported = has_shared_library(APTX_HD_ENCODER_LIB_NAME);
           break;
+        case BTAV_A2DP_CODEC_INDEX_SOURCE_APTX_ADAPTIVE:
+          // Codec aptX-HD is supported only if the device has the corresponding
+          // shared library installed.
+          supported = has_shared_library(APTX_ADAPTIVE_ENCODER_LIB_NAME);
+          break;
         case BTAV_A2DP_CODEC_INDEX_SOURCE_LDAC:
           // Codec LDAC is supported only if the device has the corresponding
           // shared library installed.
-- 
2.28.0

